packages/core/types/index.js

```diff
@@ -143,19 +143,20 @@ export type PackageJSON = {
 };
 
 export type LogLevel = 'none' | 'error' | 'warn' | 'info' | 'verbose';
+export type BuildMode = 'development' | 'production' | string;
 
 export type InitialParcelOptions = {|
   entries?: FilePath | Array<FilePath>,
   rootDir?: FilePath,
   config?: ResolvedParcelConfigFile,
   defaultConfig?: ResolvedParcelConfigFile,
-  env?: {[string]: ?string},
+  env?: {[string]: string},
   targets?: ?(Array<string> | {+[string]: TargetDescriptor}),
 
-  cache?: boolean,
+  disableCache?: boolean,
```

Comment:
For reasoning about this internally and with plugins, it might make more sense for this to be something like `enableCache`, which defaults to true, so we can avoid the double negative. This is okay though.